csrc/optimization/layout_inference.h

```diff
@@ -0,0 +1,24 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <unordered_map>
+
+#include <fusion.h>
+
+namespace nvfuser {
+
+using MemoryFormat = std::vector<int64_t>;
```

Comment:
There is a messy topic.

I was avoiding the term `StrideOrder`, because that's used in our python API. I want our python API to match what thunder's semantic of `StrideOrder` is. (which is, nhwc tensor would be written as [3, 0, 2, 1]).

Meanwhile, the format notation used in codegen would mark nhwc tensor as [0, 2, 3, 1]. The reason we want that is so that it looks more consistent with our setAllocationDomain API.
```
tv0->setAllocationDomain({tv0->axis(0), tv->axis(2), tv->axis(3), tv->axis(1)}, true);
```

Reply:
Interesting... Could you add this note to the code as a comment?